### DISCUSSION TITLE
[Bug]  helm install skywalking failing with timeout

### DISCUSSION THREAD

**Discussion by roynilanjan** [apache/skywalking#10306] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Helm Chart (apache/skywalking-kubernetes)

### What happened

```
 helm -n istio-system install skywalking   oci://ghcr.io/apache/skywalking-kubernetes/skywalking-helm   --version "0.0.0-b670c41d94a82ddefcf466d54bab5c492d88d772"   -n "${SKYWALKING_RELEASE_NAMESPACE}"   --set oap.image.tag=9.2.0   --set oap.storageType=elasticsearch   --set ui.image.tag=9.2.0 --timeout=15m
Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition
```

### What you expected to happen

Skywalking installed on Kubernetes successfully and all the pods running .

### How to reproduce

```
helm -n istio-system install skywalking   oci://ghcr.io/apache/skywalking-kubernetes/skywalking-helm   --version "0.0.0-b670c41d94a82ddefcf466d54bab5c492d88d772"   -n "${SKYWALKING_RELEASE_NAMESPACE}"   --set oap.image.tag=9.2.0   --set oap.storageType=elasticsearch   --set ui.image.tag=9.2.0 --timeout=15m
Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition 
```

### Anything else

I am trying to install skywalking on a clean k3s cluster . Every time this issue is occuring.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
The log says there is waiting for conditions. So you should.check which conditions.
Posting this simple message is not.helpful if you want a community to help.

**Reply by roynilanjan** [reply]
Since nothing is installed on skywalking namespace how to check for the condition ? Please let me know.

**Reply by kezhenxu94** [reply]
Why do you expect something installed in skywalking namespace when you specify to install in `istio-system` namespace?

```
helm -n istio-system install skywalking ...
```

**Comment by kezhenxu94** [comment]
Check the pods status and logs in `istio-system` namespace, one common case of this error is that your ES server failed to spin up, resources insufficient or whatever reasons, please check logs first.

**Reply by roynilanjan** [reply]
Nothing is getting installed on istio-system namespace as well. Please see below output
k get all -n istio-system
No resources found in istio-system namespace

**Reply by roynilanjan** [reply]
I just tried the following command :
 helm install "${SKYWALKING_RELEASE_NAME}" ${REPO}/skywalking -n "${SKYWALKING_RELEASE_NAMESPACE}" \
>   --set oap.image.tag=9.2.0 \
>   --set oap.storageType=elasticsearch \
>   --set ui.image.tag=9.2.0
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"

**Reply by kezhenxu94** [reply]
Then please check your command, you specified `-n` options twice, check what's the value of `${SKYWALKING_RELEASE_NAMESPACE}`, you might install in another namespace. Anyway this should be a very obvious issue if you find the logs. Don't expect a step-by-step troubleshooting guide from the community

**Reply by roynilanjan** [reply]
This is my complete set of commands I used :

export SKYWALKING_RELEASE_VERSION=4.3.0
export SKYWALKING_RELEASE_NAMESPACE=istio-system
export REPO=skywalking
 helm install "${SKYWALKING_RELEASE_NAME}" ${REPO}/skywalking -n "${SKYWALKING_RELEASE_NAMESPACE}" \
>   --set oap.image.tag=9.2.0 \
>   --set oap.storageType=elasticsearch \
>   --set ui.image.tag=9.2.0
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
Regarding this issue i see a thread here https://github.com/projectcalico/calico/issues/4570 
My version of kubernetes is 25, both client and server.

I am not expecting step by step guide here. Since I am new to  skywalking I need some clue to proceed.

**Reply by kezhenxu94** [reply]
Your kubernetes version is too old that ElasticSearch Helm Chart doesn't support. Try upgrading the kubernetes version or use an external ES server

**Comment by amrih-pg** [comment]
@roynilanjan you could try lowering down the replicas for es and oap if this cluster is not for production

```
helm install "${SKYWALKING_RELEASE_NAME}" \
  oci://registry-1.docker.io/apache/skywalking-helm \
  --version "${SKYWALKING_RELEASE_VERSION}" \
  -n "${SKYWALKING_RELEASE_NAMESPACE}" \
  --set oap.image.tag=9.2.0 \
  --set oap.storageType=elasticsearch \
  --set ui.image.tag=9.2.0 \
  --set elasticsearch.replicas=1 \
  --set oap.replicas=1
```

variables:
```
export SKYWALKING_RELEASE_VERSION=4.4.0  # change the release version according to your need
export SKYWALKING_RELEASE_NAME=skywalking  # change the release name according to your scenario
export SKYWALKING_RELEASE_NAMESPACE=default  # change the namespace to where you want to install SkyWalking
```

Output:
```
NAME: skywalking
LAST DEPLOYED: Wed Apr 12 14:33:28 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
NOTES:
************************************************************************
*                                                                      *
*                 SkyWalking Helm Chart by SkyWalking Team             *
*                                                                      *
************************************************************************

Thank you for installing skywalking-helm.

Your release is named skywalking.

Learn more, please visit https://skywalking.apache.org/

Get the UI URL by running these commands:
  echo "Visit http://127.0.0.1:8080 to use your application"
  kubectl port-forward svc/skywalking-skywalking-helm-ui 8080:80 --namespace default
#################################################################################
######   WARNING: Persistence is disabled!!! You will lose your data when   #####
######            the SkyWalking's storage ES pod is terminated.            #####
#################################################################################
```

**Reply by ghost** [reply]
>   oci://registry-1.docker.io/apache/skywalking-helm \
>   --version "${SKYWALKING_RELEASE_VERSION}" \
>   -n "${SKYWALKING_RELEASE_NAMESPACE}" \
>   --set oap.image.tag=9.2.0 \
>   --set oap.storageType=elasticsearch \
>   --set ui.image.tag=9.2.0 \
>   --set elasticsearch.replicas=1 \
>   --set oap.replicas=1
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/shell/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/shell/.kube/config
Error: failed to download "oci://registry-1.docker.io/apache/skywalking-helm" at version "4.4.0" (hint: running `helm repo update` may help)

**Comment by amrih-pg** [comment]
Another way to see if the resources are up and running is to `get all -n default` to see which pod is having issues

So I believe `pod/skywalking-oap-init-nxhm6` pod is only meant to perform some initialization scripts, thus the status is completed and no pod is running.

```
$ kubectl get all -n default        
                                                                                                                                                                        
NAME                                                 READY   STATUS      RESTARTS   AGE
pod/elasticsearch-master-0                           1/1     Running     0          6m27s
pod/skywalking-oap-init-nxhm6                        0/1     Completed   0          6m27s
pod/skywalking-skywalking-helm-oap-bbc6fffd5-j6k7w   1/1     Running     0          6m27s
pod/skywalking-skywalking-helm-ui-58b9fd58fc-6lnwz   1/1     Running     0          6m27s

NAME                                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)               AGE
service/elasticsearch-master             ClusterIP   10.96.227.96    <none>        9200/TCP,9300/TCP     6m27s
service/elasticsearch-master-headless    ClusterIP   None            <none>        9200/TCP,9300/TCP     6m27s
service/kubernetes                       ClusterIP   10.96.0.1       <none>        443/TCP               7m11s
service/skywalking-skywalking-helm-oap   ClusterIP   10.96.226.109   <none>        11800/TCP,12800/TCP   6m27s
service/skywalking-skywalking-helm-ui    ClusterIP   10.96.139.81    <none>        80/TCP                6m27s

NAME                                             READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/skywalking-skywalking-helm-oap   1/1     1            1           6m27s
deployment.apps/skywalking-skywalking-helm-ui    1/1     1            1           6m27s

NAME                                                       DESIRED   CURRENT   READY   AGE
replicaset.apps/skywalking-skywalking-helm-oap-bbc6fffd5   1         1         1       6m27s
replicaset.apps/skywalking-skywalking-helm-ui-58b9fd58fc   1         1         1       6m27s

NAME                                    READY   AGE
statefulset.apps/elasticsearch-master   1/1     6m27s

NAME                            COMPLETIONS   DURATION   AGE
job.batch/skywalking-oap-init   1/1           2m10s      6m27s
```

**Comment by mr-vardhan24** [comment]
hello!
I want to deploy skywalking with the backend of the timescale. in Kubernetes v.25 
I have tried the steps that followed in the below link but i am not able to deploy, Please share if someone has the proper approach to deploy this. 
https://github.com/apache/skywalking-helm
